### PR TITLE
Mining-related fixups for 0.13.0

### PR DESCRIPTION
**Pull request by sdaftuar**
My suggestion for how to address the first three items mentioned in #8294:
- Add size-accounting to `addPackageTxs()`, and use that instead of `addScoreTxs()` even when `-blockmaxsize` is in use.
- Bugfix: prevent `addPackageTxs()`  from selecting witness transactions before segwit activation, even if mempool policy allows witness transactions in.
- Remove `addScoreTxs()`, which is no longer used anywhere.
- Remove `-blockminsize` command line option, which has no effect on the mining behavior anymore.

I can drop the last two commits if we decide to go a different direction.

### CODE REVIEW

**Comment by sdaftuar** [comment]
I added a warning if `-blockminsize` is given on the command line, as we discussed on IRC last week, so this PR is now ready for review.

@laanwj I left the string for the warning untranslated because we're past the string freeze for translations for 0.13.0, but I wasn't sure if that was the right thing to do?  Not sure what your preference is for how we display the warning.

**Comment by sipa** [comment]
ACK c1d61fbd080bcc29589b8d467df98efb7e89d231

Nice to see AddScoreTx disappear. This also means that the existing mining tests now all test the only remaining algorithm.

File: src/init.cpp [hunk 1]
```diff
@@ -453,7 +453,6 @@ std::string HelpMessage(HelpMessageMode mode)
 
     strUsage += HelpMessageGroup(_("Block creation options:"));
     strUsage += HelpMessageOpt("-blockmaxcost=<n>", strprintf(_("Set maximum block cost (default: %d)"), DEFAULT_BLOCK_MAX_COST));
```
Comment (MarcoFalke) [inline src/init.cpp]: Is this done in this pull?

ACTION: rename blockmaxcost to blockmaxweight (wumpus, 19:25:14)
http://www.erisian.com.au/meetbot/bitcoin-core-dev/2016/bitcoin-core-dev.2016-07-14-19.00.html

Comment (laanwj) [inline src/init.cpp]: See #8354

**Pull request merged**